mon-pix/app/components/challenge-item-generic.js

```diff
@@ -62,29 +51,15 @@ export default class ChallengeItemGeneric extends Component {
     }
   }
 
-  _start() {
-    this._elapsedTime = 0;
-    this._tick();
-  }
-
-  _tick() {
-    if (ENV.APP.isChallengeTimerEnable) {
-      const timer = later(this, function() {
-        const elapsedTime = this._elapsedTime;
-        this._elapsedTime = elapsedTime + 1;
-        if ((this._elapsedTime - this.args.challenge.timer) >= 0) {
-          this.hasChallengeTimedOut = true;
-        } else {
-          this._tick();
-        }
-      }, 1000);
-
-      this._timer = timer;
-    }
+  @action
+  updateChallengeTimedOut() {
+    this.hasChallengeTimedOut = true;
   }
 
   @action
-  validateAnswer() {
+  validateAnswer(event) {
+    event.preventDefault();
```

Comment:
**question:** À quoi sert le `event.preventDefault();` ?

Reply:
Avec l'ancienne syntaxe, le `preventDefault` était fait implicitement.
Maintenant il faut le rajouter explicitement.

> The {{action}} modifier calls event.preventDefault(); and prevents page reload for you. In Octane, you express the intent to prevent the default behavior.

(https://crunchingnumbers.live/2019/12/23/rewriting-apps-in-ember-octane/)

Reply:
La documentation de conversion officielle n'en fait pas mention. https://ember-learn.github.io/ember-octane-vs-classic-cheat-sheet/
Si on l'enlève on a la page qui se reload du coup ?

Reply:
Non ça ne marchait pas, c'est pour cela que j'ai dû me reneigner ailleurs.

Reply:
C'est potentiellement un smell quelque part ou juste au niveau des tests mais ok. Le ticket devient long

Reply:
J'ai trouvé une explication ici: https://github.com/mike-north/ember-octane-workshop/blob/master/notes/07-first-action.md

> The default behavior of a form submit is to make a GET request to the current URL w/ form data serialized into queryParams. This means our single-page app is reloaded -- not what we want. We need to prevent this default behavior -- any ideas as to how we might do this?
Update your event handler to use Event.preventDefault();, so that the default form behavior is avoided.

Reply:
C’est intéressant merci pour cette découverte

Potentiellement ça révèle que :
- soit notre submit n’est pas fait comme un submit
- soit notre bouton n’est pas réellement un submit